modules/social_features/social_core/src/PathProcessor/SecretFiles.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\social_core\PathProcessor;
+
+use Drupal\Core\PathProcessor\InboundPathProcessorInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Defines a path processor to rewrite secret file URLs.
+ *
+ * As the route system does not allow arbitrary amount of parameters convert
+ * the file path to a parameter on the request.
+ *
+ * See https://www.drupal.org/project/drupal/issues/2741939.
+ */
+class SecretFiles implements InboundPathProcessorInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processInbound($path, Request $request) {
```

Comment:
This will still require a Drupal bootstrap. What is the time taken?

Reply:
The path processor exists because contrary to Symfony, Drupal's routing system fails at having a single path segment which may be separated by `/` so the PathProcessor is only invoked when the request actually reaches Drupal. 

[As you measured](https://github.com/goalgorilla/open_social/pull/3377#issuecomment-1593259387) the requests that still reach Drupal will still require a bootstrap and 40 requests at a time can cause a stampede. However, some of the access checking which was happening for private files on-request time was very slow itself. The current system only validates the URL which is already significantly quicker.

Importantly the controller provides caching information for the URLs which ensures that subsequent requests don't hit Drupal, so the stampede only occurs once, whereas with private files it occurs multiple times.

In the original plan there was the idea of being able to take the code from `SecretFileController` and move it into a non-Drupal PHP script so that we could avoid the bootstrap altogether. Unfortunately the image style system makes that impossible, so until we outsource that from Drupal to a separate system we can't bypass Drupal there yet.